SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+        secedit.exe /export /cfg "$($tmp)" 
+        $c = ""
+        $c = Get-Content -Path $tmp
+        $currentSetting = ""
+        foreach ($s in $c) {
+            if ( $s -like "SeManageVolumePrivilege*") {
+                $x = $s.split("=", [System.StringSplitOptions]::RemoveEmptyEntries)
+                $currentSetting = $x[1].Trim()
+            }
+        }
+
+
+        if ( $currentSetting -notlike "*$($sidstr)*" ) {
+            Write-Message -Level Verbose -Message "Modify Setting ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+       
+            if ( [string]::IsNullOrEmpty($currentSetting) ) {
+                $currentSetting = "*$($sidstr)"
+            }
+            else {
+                $currentSetting = "*$($sidstr),$($currentSetting)"
+            }
+       
+            Write-Message -Level Verbose -Message "$currentSetting"
+       
+            $outfile = @"
+[Unicode]
+Unicode=yes
+[Version]
+signature="`$CHICAGO`$"
+Revision=1
+[Privilege Rights]
+SeManageVolumePrivilege = $($currentSetting)
+"@
+       
+            $tmp2 = ""
+            $tmp2 = [System.IO.Path]::GetTempFileName()
+       
+       
+            Write-Message -Level Verbose -Message "Import new settings to Local Security Policy" -ForegroundColor DarkCyan
+            $outfile | Set-Content -Path $tmp2 -Encoding Unicode -Force
+            #notepad.exe $tmp2
+            Push-Location (Split-Path $tmp2)
+       
+            try {
+                secedit.exe /configure /db "secedit.sdb" /cfg "$($tmp2)" /areas USER_RIGHTS 
+                #Write-Message -Level Verbose -Message "secedit.exe /configure /db ""secedit.sdb"" /cfg ""$($tmp2)"" /areas USER_RIGHTS "
+            }
+            finally {  
+                Pop-Location
+            }
+        }
+        else {
+            Write-Message -Level Verbose -Message "NO ACTIONS REQUIRED! Account already in ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+        }
+        Write-Message -Level Verbose -Message "Done." -ForegroundColor DarkCyan 
+    }
+
+    
+
+    #Now configure the right amount of TempDB files.
+
+    $val = 1
+
+    WHILE ($val -ne $NumberOfCores) {
+        $sqlM = 'ALTER DATABASE tempdb ADD FILE ( NAME = N''tempdev' + $val + ''', FILENAME = N''' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\DATA\tempdev' + $val + '.ndf'' , SIZE = 64MB , FILEGROWTH = 64MB)'
+        Invoke-Sqlcmd -Database master -Query $sqlM
+
+        $val++
+    }
+
+    #And make sure the standard one has the same configuration as the new ones to make sure the parallelism works
+    $sql = @'
+ALTER DATABASE TempDB   
+MODIFY FILE  
+(NAME = tempdev,  
+SIZE = 64MB, FILEGROWTH = 64MB);  
+GO  
+'@
+
+    Invoke-Sqlcmd -Database TempDB -Query $sql
+
+    #Turn off SA, primary break-in point of the naughty users
+
+    $sql = 'ALTER LOGIN sa DISABLE'
```

Comment:
I know it's best practice, but this should be left up to the user. Also note that Invoke-Sqlcmd is a command that comes from SqlServer. Our equivalent is `Invoke-DbaQuery`

Reply:
I've been thinking about this one for a while before. Should it be included or not, because in the settings i'm asking for a password and at the end i'm disabeling the thing. I can't really make a good case for it, so it's removed.